src/components/WorldMap/WorldMap.jsx

```diff
@@ -0,0 +1,295 @@
+  }
+
+
+  /**
+   * Update viewable index.
+   */
+  updateViewable() {
+    const { geoData } = this.props;
+
+    if (!geoData || geoData.features.length === 0) {
+      return;
+    }
+
+    this.viewableIndex += 1;
+
+    this.updatePoints();
+
+    if (this.viewableIndex >= geoData.features.length) {
+      this.timer.stop();
+    }
+  }
+
+  /**
+   * Redraw points on update or zoom
+   */
+  updatePoints() {
+    const { geoData, servers } = this.props;
+    // const { viewableIndex } = this.state;
+
+    if (!geoData || geoData.features.length === 0) {
+      return;
+    }
+
+    const bounds = this.path.bounds(geoData);
+    const topLeft = bounds[0];
+    const bottomRight = bounds[1];
+
+    this.svg.attr('width', bottomRight[0] - topLeft[0])
+      .attr('height', bottomRight[1] - topLeft[1])
+      .style('left', `${topLeft[0]}px`)
+      .style('top', `${topLeft[1]}px`);
+
+    this.g.attr('transform', `translate(${-topLeft[0]},${-topLeft[1]})`);
+
+    const pointScale = d3.scaleSqrt()
+      .domain([0, 100])
+      .range([2, 18])
+      .clamp(true);
+
+      // CLIENTS
+    this.path.pointRadius((d) => pointScale(d.properties.data.download_speed_mbps));
+
+    const viewable = geoData.features.slice(0, this.viewableIndex);
+
+    const points = this.g.selectAll('.client')
+      .data(viewable, (d) => d.id);
+
+    const pointsE = points.enter()
+      .append('path')
+      .classed('client', true)
+      .style('fill-opacity', 0.0);
+
+    pointsE.transition()
+      .duration(500)
+      .style('fill-opacity', 0.5);
+
+    points
+      .style('fill-opacity', 0.5);
+
+    points.exit().remove();
+
+    points.merge(pointsE)
+      .attr('d', this.path)
+      .style('fill', 'black');
+
+
+      // LINES
+
+    this.path.pointRadius(3);
+    const minLineIndex = Math.max(0, viewable.length - 50);
+
+    const lineData = viewable.slice(minLineIndex, viewable.length);
+
+    const lines = this.g.selectAll('.line')
+      .data(lineData, (d) => d.id);
+
+    const linesE = lines.enter()
+      .append('path')
+      .classed('line', true)
+      .style('stroke', 'black')
+      .style('fill', 'none')
+      .style('stroke-opacity', 0.5);
+
+    linesE
+      .call(transitionLine);
+
+    lines.exit()
+      .transition()
+      .duration(200)
+      .style('stroke-opacity', 0.0)
+      .remove();
+
+    lines.merge(linesE)
+      .attr('d', (d) => this.path(pointsToLine([d.properties.clientPos, d.properties.serverPos])))
+      .style('stroke', 'black');
+
+
+    // SERVERS
+    this.path.pointRadius(3);
+    const server = this.g.selectAll('.server')
+      .data(servers, (d) => d.id);
+
+    const serverE = server.enter()
+      .append('path')
+      .classed('server', true)
+      .style('fill-opacity', 0.5);
+
+    server.merge(serverE)
+      .attr('d', this.path);
+  }
+
+  /**
+   * Render
+   */
+  render() {
+    const styles = { width: '100%', height: '600px' };
```

Comment:
Does width 100% do anything? It's a display: block div, shouldn't that be the default?